lib/logproto/logproto-framed-server.c

```diff
@@ -99,6 +101,9 @@ log_proto_framed_server_fetch_data(LogProtoFramedServer *self, gboolean *may_rea
   if (!(*may_read))
     return FALSE;
 
+  if (self->fetch_counter++ >= MAX_FETCH_COUNT)
+    return FALSE;
```

Comment:
I think this is not necessary.
I just read about this yesterday, every edge-triggered server works like this: they have to read until EAGAIN is returned, and server programs usually do not deal with the possibility of starvation on this level. 

In theory, starvation is possible in case of super-fast senders, but this is not a real-world scenario in my opinion.
The same question has been asked on the nginx forum a few years ago:
https://forum.nginx.org/read.php?29,250351,250360#msg-250360

When a in-memory `loopback` interface is used, it might be possible to reproduce a starvation problem, but in case of real network connections or files, the order of magnitude of reading from a memory buffer is different than the speed of those devices, so the problem does not really exist (unless the CPU has a very high load).

Reply:
However, if we really want to care with this case, our other proto server implementations have to be adjusted as well.

Fortunately, epoll uses round robin, so the concept will actually work:
> If more than maxevents file descriptors are ready when epoll_wait()
       is called, then successive epoll_wait() calls will round robin
       through the set of ready file descriptors.  This behavior helps avoid
       starvation scenarios, where a process fails to notice that additional
       file descriptors are ready because it focuses on a set of file
       descriptors that are already known to be ready.

Note: We have `log-msg-size` and `log-iw-size`, they can avoid starvation together with our flow-control mechanism even if `flags(flow-control)` is not set (but this is on a different level).

Reply:
`MAX_FETCH_COUNT` adds back the old try `try_read` logic in a much cleaner way. Removing this patch did not really change our performance numbers, so I'm approving the PR.